### PR TITLE
WRO-8316: Update samples repo with latest enact dependencies.

### PR DESCRIPTION
**Pull request by alexandrumorariu**
Enact-DCO-1.0-Signed-off-by: Alexandru Morariu alexandru.morariu@lgepartner.com

### Checklist

* [X] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [X] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [X] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Updated the versions for Enact, Sandstone, Moonstone and React.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Versions updated to: 
-react 18.2.0
-enact 4.5.0
-sandstone 2.5.0
-moonstone 4.5.0
-agate 2.0.0 (all the agate samples were updated already to version 2.0)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRO-8316

### Comments

### CODE REVIEW

File: agate/all-samples/package.json [hunk 1]
```diff
@@ -30,18 +30,18 @@
...
-    "@enact/ui": "^4.5.0-rc.1",
-    "@enact/webos": "^4.5.0-rc.1",
+    "@enact/core": "^4.5.0",
+    "@enact/i18n": "^4.5.0",
+    "@enact/spotlight": "^4.5.0",
+    "@enact/ui": "^4.5.0",
+    "@enact/webos": "^4.5.0",
     "@reduxjs/toolkit": "^1.8.1",
     "ilib": "^14.14.0",
     "immer": "^9.0.12",
```
Comment (daniel-stoian-lgp) [inline agate/all-samples/package.json:40]: i think immer dependency can be removed now. check this comment for ht ereason it exists now https://github.com/enactjs/samples/pull/163#discussion_r854804539

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "samples",
-  "version": "2.0.0",
+  "version": "2.5.0",
```
Comment (daniel-stoian-lgp) [inline package.json:3]: nice catch

**Review by daniel-stoian-lgp — COMMENTED** [review]
only 1 comment

**Review by daniel-stoian-lgp — APPROVED** [review]

File: agate/all-samples/package.json (hunk 1 above)
Comment (alexandrumorariu) [reply agate/all-samples/package.json]: Done

**Comment by sjro** [comment]
LGTM

**Pull request merged**